ChristopherPenaVelez/kmq
# Add scoreboard feature for 5-round quiz

What’s New:
- Implemented a “scoreboard” system that saves final quiz scores to the database and displays the top 5 scores.
- Added a Score model in Prisma, including a relationship with User.
- Created score.ts route for POST /api/score (to save scores) and GET /api/score/top (to retrieve top 5).
- Updated quiz/page.tsx in the frontend to submit final scores and fetch the scoreboard when the 5-round quiz finishes.

Authentication:
- Created a requireAuth middleware that decodes the JWT cookie and attaches req.user.
- Ensures only authenticated users can post scores.

Why:
- Allows players to see their final score on the leaderboard, adding a competitive element.

How to Test:
1. Log in as a user.
2. Do the 5-round quiz and finish.
3. Check if the final score is posted to the scoreboard.
4. Visit /api/score/top or look at the scoreboard displayed in the final quiz screen.

Notes:
- Next steps might be adding a dedicated leaderboard page or further styling.
